openforcefield/typing/engines/smirnoff/forcefield.py

```diff
@@ -530,35 +529,40 @@ def get_handler(self, tagname, handler_kwargs=None):
         ------
         KeyError if there is no ParameterHandler for the given tagname
         """
-
+        # If there are no kwargs for the handler, initialize handler_kwargs as an empty dict
         if handler_kwargs is None:
             handler_kwargs = dict()
 
-        handler = None
-        if tagname in self._parameter_handlers:
-            # If handler already exists, make sure it is compatible
-            handler = self._parameter_handlers[tagname]
-            handler.check_handler_compatibility(handler_kwargs)
-        elif tagname in self._parameter_handler_classes:
-            new_ph_class = self._parameter_handler_classes[tagname]
-            handler = self.register_parameter_handler(new_ph_class,
-                                                      handler_kwargs)
-
-        if handler is None:
-            msg = "Cannot find a registered parameter handler for tag '{}'\n".format(
+        # Ensure that the ForceField has a ParameterHandler class registered that can handle this tag
+        if tagname in self._parameter_handler_classes:
+            ph_class = self._parameter_handler_classes[tagname]
+        else:
+            msg = "Cannot find a registered parameter handler class for tag '{}'\n".format(
                 tagname)
-            msg += "Registered parameter handlers: {}\n".format(
-                self._parameter_handlers.keys())
+            msg += "Known parameter handler class tags are {}".format(self._parameter_handler_classes.keys())
             raise KeyError(msg)
 
-        return handler
+        # Initialize a new instance of this parameter handler class with the given kwargs
+        new_handler = ph_class(**handler_kwargs,
+                               permit_cosmetic_attributes=permit_cosmetic_attributes)
 
-    def get_io_handler(self, io_format):
-        """Retrieve the parameter handlers associated with the provided tagname.
+        if tagname in self._parameter_handlers:
+            # If a handler of this class already exists, ensure that the two handlers encode compatible science
+            old_handler = self._parameter_handlers[tagname]
+            # If no handler kwargs were provided, skip the compatibility check
+            if handler_kwargs != {}:
```

Comment:
This may be harmless, but isn't `handler_kwargs = None` when handler kwargs are not provided?

Reply:
If None, it gets set to `{}` [a few lines above](https://github.com/openforcefield/openforcefield/pull/291/files/7f88b81dbd54dc76259b2f9cfbd5804b5b3e2049#diff-150e97b6404d62dc5d95291deeeae080R533)

This is because handlers like `ToolkitAM1BCCHandler` will have an empty dict as their handler kwargs (as a consequence of how `XMLParameterIOHandler` interprets elements with no attributes or children).

This function is also how people who want to access ForceField parameters will get to them (`ForceField.get_handler('Bonds')`). I vaguely remember that setting a default argument to a mutable object in dangerous in Python, so it's safer to set it to `None`, instead of `{}`.

Reply:
Ah! Gotcha. I missed that bit.